### PR TITLE
Check if application metrics are intended to be disabled

### PR DESCRIPTION
**Pull request by kevin-lindsay-1**
Simple attempt at fixing https://github.com/istio/istio/issues/34388

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[x] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by howardjohn** [comment]
SGTM. @douglas-reid any concern?

**Comment by kevin-lindsay-1** [comment]
@googlebot I signed it!

File: pilot/cmd/pilot-agent/status/server.go [hunk 1]
```diff
@@ -193,17 +193,19 @@ func NewServer(config Options) (*Server, error) {
...
-			s.prometheus.Path = "/metrics"
-		}
-		if s.prometheus.Port == "" {
-			s.prometheus.Port = "80"
-		}
-		if s.prometheus.Port == strconv.Itoa(int(config.StatusPort)) {
-			return nil, fmt.Errorf("invalid prometheus scrape configuration: "+
-				"application port is the same as agent port, which may lead to a recursive loop. "+
-				"Ensure pod does not have prometheus.io/port=%d label, or that injection is not happening multiple times", config.StatusPort)
+		if (&prom.Scrape != "false") {
```
Comment (bianpengyuan) [inline pilot/cmd/pilot-agent/status/server.go:196]: nit: can we make it `if prom.Scrape != "false"`?

Comment (kevin-lindsay-1) [reply pilot/cmd/pilot-agent/status/server.go:196]: Haha I thought the exact same thing; I don't write a ton of Go right now, and I was using the GH editor for this basic PR; 1 sec.

Comment (kevin-lindsay-1) [reply pilot/cmd/pilot-agent/status/server.go]: I really hope I don't have to figure out how to squash this in the GH UI lol.

**Comment by kevin-lindsay-1** [comment]
@bianpengyuan do you know why `tide` still hasn't run? I'd like to get all my ✅  taken care of.

**Review by bianpengyuan — APPROVED** [review]

**Comment by bianpengyuan** [comment]
tide will unblock when PR is approved. Thanks!

**Pull request merged**